tests/testthat/test-exclusions.R

```diff
@@ -198,4 +208,30 @@ test_that("it excludes properly", {
   }
 })
 
+test_that("it excludes properly by dir", {
+  read_settings(NULL)
```

Comment:
I'm afraid I don't understand what's happening in these tests. Should the user need to use `read_settings()` before using `lint_dir`? Could you add some comments to explain what is being setup and what is being tested when these tests run, please.